### ISSUE TITLE
Need to add HDF5 format for instrument config

### ISSUE THREAD

**Issue by joelvbernier** [HEXRD/hexrd#239]
This is necessary (or another non-yaml format) in cases where detector panel buffers are arrays; these *DO NOT* play well with yaml!  We can do something really simple like what is done in `instrument.GrainDataWriter_h5`
```
fid = h5py.File("config_Ceria_run292.hdf5", 'w')
instr_grp = fid.create_group('instrument')
icfg = instr.write_config()
unwrap_dict_to_h5(instr_grp, icfg, asattr=False)
fid.close()
```
There are a few issues to address first:

- [x] if dumping to yaml format, panel buffers must be removed (or set to [0., 0.]) in `HEDMInstrument.write_config()`
- [x] if dumping to HDF5, `HEDMInstrument.write_config()` must not covert array fields to lists
- [x] need to add machinery to `HEDMInstrument.__init__()` to handle instantiation from HDF5
- [x] must sync up with [hexrdgui](https://github.com/hexrd/hexrdgui)

**Comment by joelvbernier** [comment]
@cjh1 -- I want this to replace the directory-style config I/O

**Comment by joelvbernier** [comment]
Closed with #241 and hexrdgui [#869](https://github.com/HEXRD/hexrdgui/pull/869).
